File: backend/src/main/java/com/back/back9/domain/user/dto/UserWithUsernameDto.java
package com.back.back9.domain.user.dto;

import com.back.back9.domain.user.entity.User;

import java.time.LocalDateTime;

public record UserWithUsernameDto(
        Long id,
        String username,
        String userLoginId,
        LocalDateTime createdAt,
        LocalDateTime modifiedAt
) {
    public UserWithUsernameDto(Long id, String username, String userLoginId, LocalDateTime createdAt, LocalDateTime modifiedAt) {
        this.id = id;
        this.username = username;
        this.userLoginId = userLoginId;
        this.createdAt = createdAt;
        this.modifiedAt = modifiedAt;
    }
    public UserWithUsernameDto(User user) {
        this(
                user.getId(),
                user.getUsername(),
                user.getUserLoginId(),
                user.getCreatedAt(),
                user.getModifiedAt()
        );
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/user/dto/UserWithUsernameDto.kt
package com.back.back9.domain.user.dto

import com.back.back9.domain.user.entity.User
import java.time.LocalDateTime

data class UserWithUsernameDto(
    val id: Long?,
    val username: String,
    val userLoginId: String,
    val role: String,
    val createdAt: LocalDateTime?,
    val modifiedAt: LocalDateTime?
) {
    constructor(user: User) : this(
        id = user.id,
        username = user.username,
        userLoginId = user.userLoginId,
        role = user.role.name,
        createdAt = user.createdAt,
        modifiedAt = user.modifiedAt
    )
}